cirq/study/flatten_expressions.py

```diff
@@ -337,7 +337,10 @@ def transform_sweep(self,
             sweep: The sweep to transform.
         """
         sweep = sweepable.to_sweep(sweep)
```

Comment:
Is there a reason the functionality of  `dict_list_to_sweep` isn't in `sweepable.to_sweep`.  When I read the `to_sweep` code it looks to me like it is trying to do the conversion you are doing (maybe it is failing), but that we aren't supporting ListSweep in api.v2.sweeps.

Reply:
I moved this functionality into api.v2.sweeps.  Can you take another look?